src/test/scala/treadle/ClockedManuallySpec.scala

```diff
@@ -0,0 +1,150 @@
+      |    out_direct <= in1
+      |
+      |    reg1 <= in1
+      |    out_from_reg <= reg1
+      |
+      |    reg2 <= in1
+      |    out_from_neg_edge_reg <= reg2
+      |
+      |    inverted_clock <= asClock(not(asUInt(clock)))
+      |
+      """.stripMargin
+
+  "Circuit can be run, just with poking and time advance" in {
+
+    val optionsManager = new TreadleOptionsManager {
+      treadleOptions = treadleOptions.copy(
+        setVerbose = false,
+        vcdShowUnderscored = true,
+        writeVCD = true
+      )
+    }
+
+    val tester = new TreadleTester(input, optionsManager)
+
+    tester.advanceTime(300)
+
+    tester.poke("clock", 0)
+    tester.poke("reset", 0)
+    tester.poke("in1", 0)
+
+    tester.expect("out_direct", 0)
+    tester.expect("out_from_reg", 0)
+    tester.expect("inverted_clock", 1)
+
+    tester.advanceTime(100)
+
+    tester.poke("in1", 3)
+
+    tester.expect("out_direct", 3)
+    tester.expect("out_from_reg", 0)
+    tester.expect("inverted_clock", 1)
+
+    tester.poke("clock", 1)
+
+    tester.expect("out_direct", 3)
+    tester.expect("out_from_reg", 3)
+    tester.expect("inverted_clock", 0)
+
+    tester.advanceTime(100)
+
+    tester.expect("out_direct", 3)
+    tester.expect("out_from_reg", 3)
+    tester.expect("inverted_clock", 0)
+
+    tester.report()
+  }
+
+  "clock up should advance registers off negative edge" in {
+
+    val optionsManager = new TreadleOptionsManager {
+      treadleOptions = treadleOptions.copy(
+        setVerbose = true,
+        vcdShowUnderscored = true,
+        writeVCD = true
+      )
+      commonOptions = commonOptions.copy(
+        targetDirName = "test_run_dir/manually_clocked/clock_up",
+        topName = "clock_up"
+      )
+    }
+
+    val tester = new TreadleTester(input, optionsManager)
+
+    tester.advanceTime(10)
+
+    tester.poke("in1", 1)
+    tester.poke("clock", 1)
+    tester.poke("reset", 0)
+
+    tester.expect("out_direct", 1)
+    tester.expect("out_from_reg", 1)
+    tester.expect("inverted_clock", 0)
+
+    tester.advanceTime(10)
+
+    tester.poke("in1", 2)
+    tester.poke("clock", 0)
+
+    tester.expect("out_direct", 2)
+    tester.expect("out_from_reg", 1)
+    tester.expect("inverted_clock", 1)
+
+    tester.advanceTime(10)
+
+    tester.poke("in1", 1)
+
+    tester.report()
+  }
+
+  "treadle should update register at right time" in {
+    val optionsManager = new TreadleOptionsManager {
+      treadleOptions = treadleOptions.copy(
+        setVerbose = true,
+        vcdShowUnderscored = true,
+        writeVCD = true,
+        callResetAtStartUp = false
+      )
+      commonOptions = commonOptions.copy(
+        targetDirName = "test_run_dir/auto_clocked/clock_up",
+        topName = "clock_up"
+      )
+    }
+
+    val tester = new TreadleTester(input, optionsManager)
+
+    tester.poke("in1", 1)
+    tester.step()
+    tester.poke("in1", 2)
+    tester.step()
+    tester.poke("in1", 1)
+
+    tester.report()
+  }
+
+}
```

Comment:
How about some failure cases as well? "updating circuit on second update of same input without clock advance"